pkg/configv2/default.go

```diff
@@ -136,3 +137,29 @@ var Default = types.Bacalhau{
 		Interval: Day,
 	},
 }
+
+const defaultBacalhauDir = ".bacalhau"
+
+// getDefaultDataDir determines the appropriate default directory for storing repository data.
+// Priority order:
+// 1. If the environment variable BACALHAU_DIR is set and non-empty, use it.
+// 3. User's home directory with .bacalhau appended.
+// 4. User-specific configuration directory with .bacalhau appended.
+// 5. If all above fail, use .bacalhau in the current directory.
+// The function logs any errors encountered during the process and always returns a usable path.
+func getDefaultDataDir() string {
+	if repoDir, set := os.LookupEnv("BACALHAU_DIR"); set && repoDir != "" {
+		return repoDir
+	}
+
+	if userHome, err := os.UserHomeDir(); err == nil {
+		return filepath.Join(userHome, defaultBacalhauDir)
+	}
+
+	if userDir, err := os.UserConfigDir(); err == nil {
+		return filepath.Join(userDir, defaultBacalhauDir)
+	}
```

Comment:
as I've mentioned before, this should not be treated as a default datadir path. Just a default for config path

Reply:
done.